assembly/src/parsers/u32_ops.rs

```diff
@@ -219,14 +277,61 @@ pub fn parse_u32revb(_span_ops: &mut Vec<Operation>, _op: &Token) -> Result<(),
 // COMPARISON OPERATIONS
 // ================================================================================================
 
-/// TODO: implement
-pub fn parse_u32eq(_span_ops: &mut Vec<Operation>, _op: &Token) -> Result<(), AssemblyError> {
-    unimplemented!()
+/// Translates u32eq assembly instruction to VM operations.
+/// Specifically we test the first two numbers to be u32 (U32SPLIT NOT ASSERT),
+/// then perform a EQ to check the equality.
+pub fn parse_u32eq(span_ops: &mut Vec<Operation>, op: &Token) -> Result<(), AssemblyError> {
+    match op.num_parts() {
+        0 => return Err(AssemblyError::missing_param(op)),
+        1 => {
+            // Check first number is u32
+            span_ops.push(Operation::U32split);
+            span_ops.push(Operation::Not);
+            span_ops.push(Operation::Assert);
+
+            span_ops.push(Operation::Swap);
+
+            // Check second number is u32
+            span_ops.push(Operation::U32split);
+            span_ops.push(Operation::Not);
+            span_ops.push(Operation::Assert);
+
+            span_ops.push(Operation::Swap);
```

Comment:
Do we need this `swap`?

Reply:
I was thinking to keep the original order, but if we're poping it anyway I don't think so